python/lsst/daf/butler/core/butlerConfig.py

```diff
@@ -0,0 +1,126 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Configuration classes specific to the Butler
+"""
+
+import os
+
+from .config import Config
+from .datastore import DatastoreConfig
+from .schema import SchemaConfig
+from .registry import RegistryConfig
+from .storageClass import StorageClassConfig
+from .utils import doImport
+
+__all__ = ("ButlerConfig",)
+
+
+class ButlerConfig(Config):
+    """Contains the configuration for a `Butler`
+
+    The configuration is read and merged with default configurations for
+    the particular classes. The defaults are read from
+    ``$DAF_BUTLER_DIR/config`` and ``$DAF_BULTER_CONFIG_PATH``. The defaults
+    are constructed by reading first the global defaults, and then adding
+    in overrides from each entry in the colon-separated
+    ``$DAF_BUTLER_CONFIG_PATH`` in reverse order such that the entries ahead
+    in the list take priority. The registry and datastore configurations
```

Comment:
Is this opposite behavior to shell `$PATH`? If so why?

Reply:
It's not meant to be the opposite order. In `$PATH` things early in the path take precedence over things later in the path, so I thought I had to reverse the order when reading in the defaults so that earlier ones in the path override defaults appearing late in the path.